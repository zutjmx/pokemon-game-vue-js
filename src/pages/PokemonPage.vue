<template>
  <h1>¿Quién es este PokéMon?</h1>
  
  <PokemonPicture :pokemonId="98" :showPokemon="true"/>

  <PokemonOptions :pokemons="pokemonArr"/>

</template>

<script>

import PokemonPicture from '@/components/PokemonPicture'
import PokemonOptions from '@/components/PokemonOptions'

import getPokemonOptions from '@/helpers/getPokemonOptions'

//console.log(getPokemonOptions());

export default {
    components: {
        PokemonPicture,
        PokemonOptions
    },
    data() {
      return {
        pokemonArr: []
      }
    },
    methods: {
      async mixPokemonArray() {
        this.pokemonArr = await getPokemonOptions()
      }
    },
    mounted() {
      this.mixPokemonArray()
    }
}
</script>
