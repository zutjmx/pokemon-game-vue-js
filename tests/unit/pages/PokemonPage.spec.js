import { shallowMount, mount }  from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from "../mocks/pokemons.mock"

describe('PokemonPage Component', () => {
    let wrapper
    //let mixPokemonArraySpy

    beforeEach(() => {
        //mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        wrapper = shallowMount(PokemonPage)
    })
    
    test('debe de hacer match con el snapshot.', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de llamar el mixPokemonArray al montar.', () => {
        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        const wrapper = shallowMount(PokemonPage)

        expect(mixPokemonArraySpy).toBeCalled()
    })

    test('debe de hacer match con el snapshot cuando cargan los pokemons.', () => {
        
        const wrapper = mount(PokemonPage,{
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('debe de mostrar los componentes de PokemonPicture y PokemonOptions', () => {

        const wrapper = shallowMount(PokemonPage,{
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        const picture = wrapper.find('pokemon-picture-stub')
        const options = wrapper.find('pokemon-options-stub')

        //PokemonPicture
        expect(picture.exists()).toBeTruthy()
        
        //PokemonOptions
        expect(options.exists()).toBeTruthy()

        expect(picture.attributes('pokemonid')).toBe('1')
        expect(options.attributes('pokemons')).toBeTruthy()

    })

    test('pruebas con checkAnswer', async () => {

        const wrapper = shallowMount(PokemonPage,{
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        await wrapper.vm.checkAnswer(1)

        expect(wrapper.find('h2').exists()).toBeTruthy()
        expect(wrapper.vm.showPokemon).toBe(true)
        expect(wrapper.find('h2').text()).toBe(`Correcto, ${pokemons[0].name}`)

        await wrapper.vm.checkAnswer(3)
        //expect(wrapper.find('h2').text()).toBe(`Ooppsss, era => ${pokemons[0].name}`)
        expect(wrapper.vm.message).toBe(`Ooppsss, era => ${pokemons[0].name}`)

    })

})