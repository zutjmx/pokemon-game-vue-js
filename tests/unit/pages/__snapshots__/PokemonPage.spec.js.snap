// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component debe de hacer match con el snapshot cuando cargan los pokemons. 1`] = `
<div>
  <h1>¿Quién es este PokéMon?</h1>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon"></div>
  <div class="options-container">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;

exports[`PokemonPage Component debe de hacer match con el snapshot. 1`] = `<h1>Espere por favor .... </h1>`;
